kratos/utilities/intersection_utilities.h

```diff
@@ -795,6 +798,76 @@ class KRATOS_API(KRATOS_CORE) IntersectionUtilities
         return false;
     }
 
+    /**
+     * @brief Computes the intersection line between two planes.
+     * @details Given two planes defined by a point and a normal for each, this function computes
+     *          the intersection line between them. The direction (tangent) of the intersection line is
+     *          computed as the cross product of the two normals. If this cross product is nearly zero,
+     *          the planes are parallel (or coincident) and no unique intersection line exists.
+     *
+     *          If the planes intersect in a unique line, a point on the line is determined using the formula:
+     *          @f[
+     *            \mathbf{x}_0 = \frac{ \left[ (d_1 \mathbf{n}_2 - d_2 \mathbf{n}_1) \times (\mathbf{n}_1 \times \mathbf{n}_2) \right] }{ \| \mathbf{n}_1 \times \mathbf{n}_2 \|^2 }
+     *          @f]
+     *          where @f$ d_1 = \mathbf{n}_1 \cdot \mathbf{p}_1 @f$ and @f$ d_2 = \mathbf{n}_2 \cdot \mathbf{p}_2 @f$.
+     *          Note that the normals need not be unit length.
+     * @param rPlanePoint1 One point on the first plane.
+     * @param rNormal1 The normal vector of the first plane.
+     * @param rPlanePoint2 One point on the second plane.
+     * @param rNormal2 The normal vector of the second plane.
+     * @param rPoint [out] A point on the intersection line (if one exists).
+     * @param rTangent [out] The direction (tangent) vector of the intersection line.
+     * @param Tolerance The tolerance to check the parallel
+     * @return 0 if the planes are parallel (or nearly parallel), 1 if the planes intersect in a unique line.
+     */
+    static int Compute2PlaneIntersection(
```

Comment:
```suggestion
    static int ComputeTwoPlaneIntersection(
```
I think this name is more aligned with the rest of Kratos. Up to you.